apps/yggdrasil/lib/yggdrasil/user.ex

```diff
@@ -25,9 +27,22 @@ defmodule Yggdrasil.User do
     |> validate_length(:password, min: 4)
     |> validate_length(:password_confirmation, min: 4)
     |> validate_confirmation(:password)
+    |> default_role
     |> hash_password
   end
 
+  def with_role(query) do
```

Comment:
Preload roles and the role_resources, role_permissions etc.

First pass I have other ideas, but for now it suffices. Ecto 2.0 will simplify a lot of this.